realm/realm-library/src/objectServer/java/io/realm/permissions/PermissionOffer.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.permissions;
+
+import java.util.Date;
+import java.util.UUID;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.realm.RealmObject;
+import io.realm.annotations.Index;
+import io.realm.annotations.PrimaryKey;
+import io.realm.annotations.Required;
+
+/**
+ * This model is used for offering permission changes to other users.
+ * It should be used in conjunction with an {@link io.realm.SyncUser}'s management Realm.
+ *
+ * @see <a href="https://realm.io/docs/realm-object-server/#permissions">Permissions description</a> for general
+ * documentation.
+ */
+public class PermissionOffer extends RealmObject {
+
+    // Base fields
+    @PrimaryKey
+    @Required
+    private String id = UUID.randomUUID().toString();
```

Comment:
It seems like a shame to initialize these things twice.  Could you do this in the no-args constructor?  There is no check for `null`s in the full constructor...